pandas/core/arrays/base.py

```diff
@@ -0,0 +1,194 @@
+"""An interface for extending pandas with custom arrays."""
+import abc
+
+import numpy as np
+
+from pandas.compat import add_metaclass
+
+
+_not_implemented_message = "{} does not implement {}."
+
+
+@add_metaclass(abc.ABCMeta)
+class ExtensionArray(object):
+    """Abstract base class for custom array types
+
+    Notes
+    -----
+    pandas will recognize instances of this class as proper arrays
+    with a custom type and will not attempt to coerce them to objects.
+
+    **Restrictions on your class constructor**
+
+        * Your class should be able to be constructed with no arguments,
+          i.e. ``ExtensionArray()`` returns an instance.
+          TODO: See comment in ``ExtensionDtype.construct_from_string``
```

Comment:
What exactly is `ExtensionArray()` supposed to return? A length zero array? This should be clarified.

I don't understand the comment about `construct_from_string`. `construct_from_sting` currently returns an instance of the dtype object, not an array.

Also, in my opinion is somewhat poor API design for an empty constructor to automatically correspond to a certain result. I would rather have an explicit `empty()` class method.

Reply:
Ignore my comments about `ExtensionArray()` . That was apparently vestigial from an earlier version of `construct_from_string`. Agreed that this would have been an awful design decision to force on subclasses :)